### PR TITLE
Branch add budget

### LINKED ISSUE #68: Add optional budget field to vendor

### PR DESCRIPTION
**Pull request by slye20**
Fixes #68

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Budget.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            budget = "0" + budget;
+        }
+
+        // Fixes "x.x"
+        if (parts[1].length() == 1) {
+            budget += "0";
+        }
+        value = budget;
+    }
+
```
Comment (weibinlim) [inline src/main/java/seedu/address/model/person/Budget.java:56]: Nice! Your checks are very extensive!

File: src/main/java/seedu/address/logic/commands/util/EditVendorDescriptor.java [hunk 2]
```diff
@@ -42,6 +44,14 @@ public Optional<Company> getCompany() {
...
     }
 
+    public void setBudget(Budget budget) {
+        this.budget = budget;
+    }
+
+    public Optional<Budget> getBudget() {
+        return Optional.ofNullable(budget);
+    }
+
```
Comment (weibinlim) [inline src/main/java/seedu/address/logic/commands/util/EditVendorDescriptor.java:54]: Good usage of Optional!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -159,6 +160,23 @@ public static Company parseCompany(String company) throws ParseException {
...
+     * @throws ParseException if the given {@code email} is invalid.
+     */
+    public static Budget parseBudget(String budget) throws ParseException {
+        if (budget == null) {
+            return null;
+        }
+        String trimmedBudget = budget.trim();
+        if (!Budget.isValidBudget(trimmedBudget)) {
+            throw new ParseException(Budget.MESSAGE_CONSTRAINTS);
+        }
```
Comment (YangQF2002) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:176]: Nice way to parse the budget

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 4]
```diff
@@ -58,22 +58,33 @@ public PersonCard(Person person, int displayedIndex) {
...
     }
+
+    /**
+     * Adds a label to the cardPaneContents with a specific style class.
+     * @param label The Label to be added to the UI.
+     */
+    public void addLabel(Label label) {
+        label.getStyleClass().add("cell_small_label");
+        cardPaneContents.getChildren().add(label);
+    }
```
Comment (YangQF2002) [inline src/main/java/seedu/address/ui/PersonCard.java:89]: Nice to use a method to extract out the commonly performed operations

File: src/test/java/seedu/address/testutil/TypicalVendors.java
Comment (YangQF2002) [inline src/test/java/seedu/address/testutil/TypicalVendors.java:1]: Nice changes to the test persons

**Pull request merged**